pkg/collector/corechecks/snmp/snmp.go

```diff
@@ -0,0 +1,141 @@
+package snmp
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"time"
+)
+
+const (
+	snmpCheckName = "snmp"
+)
+
+// Check aggregates metrics from one Check instance
+type Check struct {
+	core.CheckBase
+	config  snmpConfig
+	session sessionAPI
+	sender  metricSender
+}
+
+// Run executes the check
+func (c *Check) Run() error {
+	start := time.Now()
+
+	sender, err := aggregator.GetSender(c.ID())
+	if err != nil {
+		return err
+	}
+	c.sender = metricSender{sender: sender}
+
+	staticTags := c.config.getStaticTags()
+
+	var checkErr error
+	tags, checkErr := c.doRun(staticTags)
```

Comment:
nit: could be called `doRunAndSendMetrics`. As you send service check and performance metrics under here I wondered where the actual metrics were sent

Reply:
I think we still want to allow `doRun` to send other kind of service checks if needed.

Reply:
Sure, my point here is that having calls to send metrics under here, it can be misleading to call this `doRun` as the send metric part is split.

Reply:
Renamed to `processSnmpMetrics` since that's what we actually do: https://github.com/DataDog/datadog-agent/pull/6749/commits/923b1353803bb4311188c933382ef14f736202ce